packages/eas-cli/src/commands/update/index.ts

```diff
@@ -197,7 +217,13 @@ export default class UpdatePublish extends EasCommand {
       exp: expPossiblyWithoutEasUpdateConfigured,
       projectId,
       projectDir,
-    } = await getDynamicPublicProjectConfigAsync();
+    } = withEasEnvironmentVariablesSet
+      ? await resolveExpoConfigWithEasEnvironmentVariablesLoadedAsync({
+          environment: withEasEnvironmentVariablesSet,
+          graphqlClient,
+          getDynamicPublicProjectConfigAsync,
+        })
+      : await getDynamicPublicProjectConfigAsync();
 
     await maybeWarnAboutEasOutagesAsync(graphqlClient, [StatuspageServiceName.EasUpdate]);
 
```

Comment:
<!--__GRAPHITE_HTML_TAG_START__--><i class='graphite__hidden'>[Re: lines +239 to +240]<br/><br/></i>

<!--__GRAPHITE_HTML_TAG_END__-->do we need it for these calls as well?<!--__GRAPHITE_HTML_TAG_START__--><span class='graphite__hidden'><br/><br/>See this comment inline on <a href="https://app.graphite.dev/github/pr/expo/eas-cli/2628?utm_source=unchanged-line-comment">Graphite</a>.</span>

<!--__GRAPHITE_HTML_TAG_END__-->